extensions/ql-vscode/src/remote-queries/remote-queries-manager.ts

```diff
@@ -131,40 +131,21 @@ export class RemoteQueriesManager extends DisposableObject {
     const executionEndTime = Date.now();
 
     if (queryWorkflowResult.status === 'CompletedSuccessfully') {
-      const resultIndex = await getRemoteQueryIndex(credentials, queryItem.remoteQuery);
-      queryItem.completed = true;
-      if (resultIndex) {
-        queryItem.status = QueryStatus.Completed;
-        const queryResult = this.mapQueryResult(executionEndTime, resultIndex, queryItem.queryId);
-
-        await this.storeJsonFile(queryItem, 'query-result.json', queryResult);
-
-        // Kick off auto-download of results in the background.
-        void commands.executeCommand('codeQL.autoDownloadRemoteQueryResults', queryResult);
-
-        // Ask if the user wants to open the results in the background.
-        void this.askToOpenResults(queryItem.remoteQuery, queryResult).then(
-          noop,
-          err => {
-            void showAndLogErrorMessage(err);
-          }
-        );
-      } else {
-        void showAndLogErrorMessage(`There was an issue retrieving the result for the query ${queryItem.remoteQuery.queryName}`);
-        queryItem.status = QueryStatus.Failed;
-      }
+      await this.downloadAvailableResults(queryItem, credentials, executionEndTime);
     } else if (queryWorkflowResult.status === 'CompletedUnsuccessfully') {
       if (queryWorkflowResult.error?.includes('cancelled')) {
         // workflow was cancelled on the server
         queryItem.failureReason = 'Cancelled';
         queryItem.status = QueryStatus.Failed;
+        await this.downloadAvailableResults(queryItem, credentials, executionEndTime);
         void showAndLogInformationMessage('Variant analysis was cancelled');
```

Comment:
I don't really like my suggestion, but we should get the point across that we are downloading the results that exist.

```suggestion
        void showAndLogInformationMessage('Variant analysis was cancelled. Downloading partial results.');
```

Reply:
FWIW I don't think we necessarily need this - the usual view results notification is shown.

Reply:
Agreed that we could do with some better UI 🙈 I'll leave as-is for now, and we can follow-up with a nicer experience later. (Along with query history icons, as mentioned separately!)